modal/runner.py

```diff
@@ -292,11 +302,32 @@ async def _interactive_shell(_function: _Function, cmd: str, environment_name: s
     ```bash
     modal shell script.py --cmd /bin/bash
     ```
+
+    **kwargs will be passed into spawn_sandbox().
     """
-    _stub = _function._stub
-    _stub._add_pty_input_stream()  # TOOD(erikbern): slightly hacky
     async with _run_stub(_stub, environment_name=environment_name, shell=True):
-        await _function.shell(cmd)
+        console = Console()
+        loading_status = console.status("Starting container...")
+        loading_status.start()
+
+        sb = await _stub.spawn_sandbox("sleep", "360000", **kwargs)
+
+        for _ in range(40):
+            await asyncio.sleep(0.5)
+            resp = await sb._client.stub.SandboxGetTaskId(api_pb2.SandboxGetTaskIdRequest(sandbox_id=sb._object_id))
+            if resp.task_id != "":
+                task_id = resp.task_id
+                break
+            # else: sandbox hasn't been assigned a task yet
+        else:
+            loading_status.stop()
+            rich.print("Error: timed out waiting for sandbox to start", file=sys.stderr)
+            await sb.terminate()
```

Comment:
Don't think this is needed explicitly, since the `with _run_stub` context manager will terminate all apps on exit.